dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResourceHelper.java

```diff
@@ -453,7 +453,9 @@ private Template checkoutTemplate(final IHTMLPage page, final User user, final P
                 .anyMatch(pageVersion -> !page.getIdentifier().equals(pageVersion.getIdentifier()) ||
                         !((HTMLPageAsset) page).getVariantId().equals(pageVersion.getVariantName()));
 
-        if (!useByAnotherPage) {
+        final Template systemTemplate = APILocator.getTemplateAPI().systemTemplate();
+
+        if (!useByAnotherPage && systemTemplate != oldTemplate) {
```

Comment:
the id of the system table is SYSTEM_TEMPLATE you can check the id, would be more accurate than memory location comparison

Reply:
done https://github.com/dotCMS/core/pull/27175/commits/c2d119ce7ddca28e4f8e1d9a8345e60dc1fc2db0